plugins/webp-uploads/picture-element.php

```diff
@@ -95,12 +95,10 @@ function webp_uploads_wrap_image_in_picture( string $image, string $context, int
 	list( $src, $width, $height ) = $image_src;
 	$size_array                   = array( absint( $width ), absint( $height ) );
 
-	foreach ( $mime_types as $image_mime_type ) {
-		$sizes = wp_calculate_image_sizes( $size_array, $src, $image_meta, $attachment_id );
-		if ( false === $sizes ) {
-			continue;
-		}
+	// Get the sizes from the IMG tag.
+	$sizes = (string) $processor->get_attribute( 'sizes' );
```

Comment:
Note that `$processor->get_attribute( 'sizes' )` here could return `null` if no attribute is present (or `true` in the case of `<img sizes>`, which is unlikely). Therefore, I recommend removing the `(string)` cast and instead when using the `$sizes` attribute to only add it to the `$picture_sources` markup if `is_string( $sizes )`.

Reply:
```suggestion
	$sizes = $processor->get_attribute( 'sizes' );
```